src/include/utils.h

```diff
@@ -375,6 +375,62 @@ pgagroal_backtrace(void);
 
 #endif
 
+/**
+ * Command parsing utils for pgagroal-cli and pgagroal-admin.
+ */
+
+#define COMMAND_UNKNOWN (-1)
+
+struct pgagroal_parsed_command
+{
+   int command_code;
+   char* args[MISC_LENGTH];
+   int arg_count;
+   int action;
+   int mode;
+   char* log_trace;
+   char error_message[MISC_LENGTH];
+};
+
+/**
+ * Struct for defining pgagroal commands.
+ * The necessary fields are marked with an ">".
+ *
+ * Fields:
+ * > command: The primary name of the command.
+ * > subcommand: The subcommand name. If there is no subcommand, it should be filled with an empty literal string.
+ * > accepted_argument_count: An array defining all the number of arguments this command accepts.
+ *    Each entry represents a valid count of arguments, allowing the command to support overloads.
+ * - default_argument: A default value for the command argument, used when no explicit argument is provided.
+ * - log_trace: A template string for logging command execution, which can include placeholders for dynamic values.
+ * > action: A value indicating the specific action.
+ * - mode: A value specifying the mode of operation or context in which the command applies.
+ * > deprecated: A flag indicating whether this command is deprecated.
+ * - deprecated_since_major: The major version number in which the command was deprecated.
+ * - deprecated_since_minor: The minor version number in which the command was deprecated.
+ * - deprecated_by: A string naming the command that replaces the deprecated command.
+ *
+ * This struct is key to extending and maintaining the command processing functionality in pgagroal,
+ * allowing for clear definition and handling of all supported commands.
+ */
+struct pgagroal_command
+{
+   const char* command;
+   const char* subcommand;
+   int accepted_argument_count[MISC_LENGTH];
+
+   char* default_argument;
+   char* log_trace;
```

Comment:
I would name this field as `log_message` or something like that. The fact that we, at the moment, log as `trace` has nothing to do with the message itself.